core/common/src/main/scala/org/typelevel/otel4s/context/propagation/ContextPropagators.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.context.propagation
+
+/** A container of the registered propagators for every supported format.
+  *
+  * @tparam Ctx
+  *   the type of the Context
+  */
+trait ContextPropagators[Ctx] {
```

Comment:
We can make it a `sealed trait`. But should we?

Reply:
you'd have to remove `ContextPropagatorsImpl` from the java-common for that, though I suppose that's not necessarily a bad thing.

if you seal it and that turns out to be inconvenient, it can always be unsealed in a minor release, so IMO go for it

Reply:
Valid point. Let's give it a try